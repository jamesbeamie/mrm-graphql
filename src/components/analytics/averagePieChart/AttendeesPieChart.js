import React, { Component } from 'react';

// import svg files
import { attendeesChart } from '../../../../src/utils/images/images';

// import styles
import '../../../../src/assets/styles/pieChartBaseStyle.scss';
import '../../../../src/assets/styles/attendeesPieChart.scss';

// Import the tip
import Tip from '../../commons/Tooltip';

/**
 * Attendees Pie chart component
 *
 * @returns {JSX}
 */
class AttendeesPieChart extends Component {
  // eslint-disable-next-line
  state = {};

  render() {
    const tip = 'Number of attendees in a booked meeting room per meeting room';
    return (
      <article className="pie-chart">
        <section className="chart-header">
          <p className="chart-title">Average No. of Attendees/ Meeting</p>
          {Tip(tip)}
        </section>
        <section className="chart-content">
          <img src={attendeesChart} alt="average meeting chart" />
          <section className="chart-details">
            <p className="attendees-first-circle">
              <span>{}</span>
              130 (Yes)
            </p>
            <p className="attendees-second-circle">
              <span>{}</span>
              20 (Maybe)
            </p>
            <p className="attendees-third-circle">
              <span>{}</span>6 (No)
            </p>
          </section>
        </section>
      </article>
    );
  }
}

export default AttendeesPieChart;
