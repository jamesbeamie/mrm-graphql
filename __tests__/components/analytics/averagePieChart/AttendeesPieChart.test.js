import React from 'react';
import { shallow } from 'enzyme';
import AttendeesPieChart from '../../../../src/components/analytics/averagePieChart/AttendeesPieChart';

describe('Attendees Pie Chart Component', () => {
  const wrapper = shallow(<AttendeesPieChart />);
  it('renders correctly from memory', () => {
    expect(shallow(<AttendeesPieChart />)).toMatchSnapshot();
  });
  it('should be able to hover', () => {
    const helpIcon = wrapper.find('img[alt="help icon"]');
    const nativeEvent = { nativeEvent: { clientX: 300, clientY: 250 } };
    helpIcon.simulate('mouseenter', nativeEvent);
    expect(wrapper).toMatchSnapshot();
    helpIcon.simulate('mouseleave', nativeEvent);
    expect(wrapper).toMatchSnapshot();
  });
});
