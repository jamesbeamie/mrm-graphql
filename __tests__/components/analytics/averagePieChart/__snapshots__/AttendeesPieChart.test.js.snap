// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attendees Pie Chart Component renders correctly from memory 1`] = `
<article
  className="pie-chart"
>
  <section
    className="chart-header"
  >
    <p
      className="chart-title"
    >
      Average No. of Attendees/ Meeting
    </p>
    <ThemedTooltippedComponent
      className=""
      composeTheme="deeply"
      mapThemrProps={[Function]}
      theme={null}
      tooltip="Number of attendees in a booked meeting room per meeting room"
      tooltipDelay={0}
      tooltipHideOnClick={true}
      tooltipPosition="top"
      tooltipShowOnClick={true}
    >
      <img
        alt="help icon"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
      />
    </ThemedTooltippedComponent>
  </section>
  <section
    className="chart-content"
  >
    <img
      alt="average meeting chart"
    />
    <section
      className="chart-details"
    >
      <p
        className="attendees-first-circle"
      >
        <span />
        130 (Yes)
      </p>
      <p
        className="attendees-second-circle"
      >
        <span />
        20 (Maybe)
      </p>
      <p
        className="attendees-third-circle"
      >
        <span />
        6 (No)
      </p>
    </section>
  </section>
</article>
`;

exports[`Attendees Pie Chart Component should be able to hover 1`] = `
<article
  className="pie-chart"
>
  <section
    className="chart-header"
  >
    <p
      className="chart-title"
    >
      Average No. of Attendees/ Meeting
    </p>
    <ThemedTooltippedComponent
      className=""
      composeTheme="deeply"
      mapThemrProps={[Function]}
      theme={null}
      tooltip="Number of attendees in a booked meeting room per meeting room"
      tooltipDelay={0}
      tooltipHideOnClick={true}
      tooltipPosition="top"
      tooltipShowOnClick={true}
    >
      <img
        alt="help icon"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
      />
    </ThemedTooltippedComponent>
  </section>
  <section
    className="chart-content"
  >
    <img
      alt="average meeting chart"
    />
    <section
      className="chart-details"
    >
      <p
        className="attendees-first-circle"
      >
        <span />
        130 (Yes)
      </p>
      <p
        className="attendees-second-circle"
      >
        <span />
        20 (Maybe)
      </p>
      <p
        className="attendees-third-circle"
      >
        <span />
        6 (No)
      </p>
    </section>
  </section>
</article>
`;

exports[`Attendees Pie Chart Component should be able to hover 2`] = `
<article
  className="pie-chart"
>
  <section
    className="chart-header"
  >
    <p
      className="chart-title"
    >
      Average No. of Attendees/ Meeting
    </p>
    <ThemedTooltippedComponent
      className=""
      composeTheme="deeply"
      mapThemrProps={[Function]}
      theme={null}
      tooltip="Number of attendees in a booked meeting room per meeting room"
      tooltipDelay={0}
      tooltipHideOnClick={true}
      tooltipPosition="top"
      tooltipShowOnClick={true}
    >
      <img
        alt="help icon"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
      />
    </ThemedTooltippedComponent>
  </section>
  <section
    className="chart-content"
  >
    <img
      alt="average meeting chart"
    />
    <section
      className="chart-details"
    >
      <p
        className="attendees-first-circle"
      >
        <span />
        130 (Yes)
      </p>
      <p
        className="attendees-second-circle"
      >
        <span />
        20 (Maybe)
      </p>
      <p
        className="attendees-third-circle"
      >
        <span />
        6 (No)
      </p>
    </section>
  </section>
</article>
`;
